staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -109,17 +109,12 @@ func New(
 	clusterName string,
 	featureGate featuregate.FeatureGate,
 ) (*Controller, error) {
-	broadcaster := record.NewBroadcaster()
-	recorder := broadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "service-controller"})
-
 	registerMetrics()
 	s := &Controller{
 		cloud:            cloud,
 		kubeClient:       kubeClient,
 		clusterName:      clusterName,
 		cache:            &serviceCache{serviceMap: make(map[string]*cachedService)},
-		eventBroadcaster: broadcaster,
-		eventRecorder:    recorder,
```

Comment:
Did the panic risk for this get fixed as well?

Reply:
This got fixed for 1.32 in https://github.com/kubernetes/kubernetes/pull/128179.

A fix for 1.31 is in https://github.com/kubernetes/kubernetes/pull/128959.